agents/vantagePro2_agent/vantage_pro2_agent.py

```diff
@@ -0,0 +1,166 @@
+
+            session.set_status('starting')
+
+            self.module = VantagePro2(self.port)
+            print("Initialized Vantage Pro2 module: {!s}".format(
+                self.module))
+
+        self.initialized = True
+
+        # Start data acquisition if requested
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Vantage Pro2 module initialized.'
+
+    def start_acq(self, session, params=None):
+        """
+        Method to start data acquisition process.
+
+        Args:
+            sample_freq (int):
+                Frquency at which weather data is sampled.
+                Defaults to sample/2 seconds, the highest frequency possible.
+
+        """
+        time.sleep(2)
+        if params is None:
+            params = {}
+
+        sample_freq = params.get('freq')
+        # If loops is None, use value passed to Agent init
+        if sample_freq is None:
+            sample_freq = self.freq
+
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running"
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'weather',
+                    'data': {}
+                }
+
+                data['data'] = self.module.weather_daq()
+                self.agent.publish_to_feed('weather_data', data)
+                time.sleep(sample_freq)
+
+            self.agent.feeds['weather_data'].flush_buffer()
+
+        return True, 'Acquisition exited cleanly.'
+
+    def stop_acq(self, session, params=None):
+        """
+        Stops acq process.
+        """
+
+        if self.take_data:
+            self.take_data = False
+            print('requested to stop taking data.')
+            return True, 'data taking succesfully halted'
+        else:
+            return False, 'acq is not currently running'
+
+
+def make_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--port', type=str,
+                        help="Path to USB node for the Vantage Pro2")
+    pgroup.add_argument('--mode', type=str, choices=['idle', 'init', 'acq'],
+                        help="Starting action for the agent.")
+    pgroup.add_argument('--freq', type=int,
+                        help="Sample frequency for weather data collection")
+    return parser
+
+
+if __name__ == "__main__":
+    parser = make_parser()
+    args = site_config.parse_args(
+        agent_class='VantagePro2Agent', parser=parser)
+    startup = False
+
+    if args.mode == 'acq':
```

Comment:
For an example of how to make the auto-acquire work, see the ls240 agent: https://github.com/simonsobs/socs/blob/015f25e9ad9b93b610971bfe93d6a1c4252df83e/agents/lakeshore240/LS240_agent.py#L322-L327 

If `startup` is a dict, it will be passed as `params` to the task on start.